pkg/deployer/deployer.go

```diff
@@ -58,56 +58,48 @@ func (d *Deployer) GetIP(cluster *clusterv1.Cluster, machine *clusterv1.Machine)
 		return scope.ClusterStatus.Network.APIServerIP.IPAddress, nil
 	}
 
-	pipsvc := network.NewService(scope)
+	pipSvc := network.NewService(scope)
+	// TODO: Remove once resourcesSvc.GetPublicIPName() has moved to network package.
+	resourcesSvc := resources.NewService(scope)
 
-	pip, err := pipsvc.GetPublicIPAddress(scope.ClusterConfig.ResourceGroup, resources.GetPublicIPName(machine))
+	pip, err := pipSvc.CreateOrUpdatePublicIPAddress(scope.ClusterConfig.ResourceGroup, resourcesSvc.GetPublicIPName(machine))
```

Comment:
The more that I think about this, I have a preference towards naming this `EnsurePublicIPAddress`. Thoughts?

Reply:
Still sticking to my comment above :stuck_out_tongue: :https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/88#discussion_r260110486

> I initially had this as `CreateOrGetPublicAddress` with separate logic for each, but we actually always want to reconcile it (update DNS name label, etc.), so I changed the method name.
> 
> I thought about `EnsurePublicIPAddress`, because it's similar to the usage in `cloud-provider-azure`, but decided on `CreateOrUpdate`, because it's more inline with the underlying Azure call (`s.scope.PublicIPAddresses.CreateOrUpdate`).